### PR TITLE
Building: Fix ACAV hammer usage and recursive scriptedEventHandler failure.

### PR DESCRIPTION
**Pull request by dijksterhuis**
well this was a FUN one.

Originally the hammer was set up to only actually destroy buildings when the progress change matched the hardcoded value of `0.5`. this would exclude the ACAV players from being able to use the hammer to destroy buildings as their progress change for the hammer was set to `0.2`.

So I changed the logic to just watch for a new building state `< 0` -- i.e. we've hit the building with the hammer enough times to make it's state go negative. It should take several hits to get to this point.

But the hammer was removing the building immediately on first hit, which made absolutely no sense whatsoever.

Turns out the hammer was causing a recursive failure in the CDLC's `VN_fnc_melee_fired` function -- where the scripted event handler set up by `fn_tool_controller_init.sqf` is called. Apparently it was returning a string value, not a boolean.

It seems the reason for this was there was no `false` at the end of the `fn_operate_hammer.sqf` script. Which meant that way back up in the scripted event handler created in `fn_tool_controller_init.sqf` the called function was returning a string -- the return value of the `building_on_hit` remote exec.

TL;DR function returns matter.

---

TODOs

- [ ] Need to think about stopping abuse from all public players as paradigm shouldn't have any knowledge of Mike Force internals -- so need to look for a custom unit trait set in `teams.hpp` for something like `canUseHammer = false` on non-ACAV public teams.

### CODE REVIEW

**Comment by dijksterhuis** [comment]
Sent into draft as i'm no longer writing code on BN Mike Force 1. Want to make available for other people to look through and see what's what.
